src/library/autodj/autodjprocessor.cpp

```diff
@@ -504,6 +504,16 @@ void AutoDJProcessor::playerPositionChanged(DeckAttributes* pAttributes,
 
 TrackPointer AutoDJProcessor::getNextTrackFromQueue() {
     // Get the track at the top of the playlist.
+    bool randomQueueEnabled = (((m_pConfig->getValueString(
+                ConfigKey("[Auto DJ]", "EnableRandomQueue")).toInt())) == 1);
+    int minAutoDJCrateTracks = m_pConfig->getValueString(
+                ConfigKey(kConfigKey, "RandomQueueMinimumAllowed")).toInt();
+    int tracksToAdd = minAutoDJCrateTracks - m_pAutoDJTableModel->rowCount();
+    // Incase we start off with < minimum tracks
+    if (randomQueueEnabled && (tracksToAdd > 0)) {
+        emit(randomTrackRequested(tracksToAdd));
```

Comment:
@daschuer This should fix the endless loop issue. Thanks for the suggestion.
I'v also declared the global variable as you asked.